SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestCharacterMovement/SpatialTestCharacterMovement.cpp

```diff
@@ -50,110 +50,108 @@ void ASpatialTestCharacterMovement::BeginPlay()
-
-		UBoxComponent* BoxComponent = Cast<UBoxComponent>(TriggerBox->GetCollisionComponent());
-		if (BoxComponent)
+	// Universal setup step to create the TriggerBox and to set the helper variable
+	AddUniversalStep(TEXT("UniversalSetupStep"), nullptr, [this](ASpatialFunctionalTest* NetTest)
 		{
-			BoxComponent->SetBoxExtent(FVector(10.0f, 1.0f, 1.0f));
-		}
-
-		TriggerBox->OnActorBeginOverlap.AddDynamic(Test, &ASpatialTestCharacterMovement::OnOverlapBegin);
-
-		Test->FinishStep();
-		});
+			bCharacterReachedDestination = false;
 
-	// The server checks if the clients received a TestCharacterMovement and moves them to the mentioned locations
-	AddServerStep(TEXT("SpatialTestCharacterMovementServerSetupStep"), 1, nullptr, [this](ASpatialFunctionalTest* NetTest) {
-		ASpatialTestCharacterMovement* Test = Cast<ASpatialTestCharacterMovement>(NetTest);
+			ATriggerBox* TriggerBox = GetWorld()->SpawnActor<ATriggerBox>(FVector(232.0f, 0.0f, 40.0f), FRotator::ZeroRotator, FActorSpawnParameters());
 
-		for (ASpatialFunctionalTestFlowController* FlowController : NetTest->GetFlowControllers())
-		{
-			if (FlowController->ControllerType == ESpatialFunctionalTestFlowControllerType::Server)
+			UBoxComponent* BoxComponent = Cast<UBoxComponent>(TriggerBox->GetCollisionComponent());
+			if (BoxComponent)
 			{
-				continue;
+				BoxComponent->SetBoxExtent(FVector(10.0f, 1.0f, 1.0f));
 			}
 
-			AController* PlayerController = Cast<AController>(FlowController->GetOwner());
-			ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
+			TriggerBox->OnActorBeginOverlap.AddDynamic(this, &ASpatialTestCharacterMovement::OnOverlapBegin);
 
-			checkf(PlayerCharacter, TEXT("Client did not receive a TestMovementCharacter"));
+			FinishStep();
+		});
 
-			if (FlowController->ControllerInstanceId == 1)
-			{
-				PlayerCharacter->SetActorLocation(FVector(0.0f, 0.0f, 50.0f));
-			}
-			else
+	// The server checks if the clients received a TestCharacterMovement and moves them to the mentioned locations
+	AddServerStep(TEXT("SpatialTestCharacterMovementServerSetupStep"), 1, nullptr, [this](ASpatialFunctionalTest* NetTest)
+		{
+			for (ASpatialFunctionalTestFlowController* FlowController : GetFlowControllers())
 			{
-				PlayerCharacter->SetActorLocation(FVector(100.0f, 300.0f, 50.0f));
+				if (FlowController->ControllerType == ESpatialFunctionalTestFlowControllerType::Server)
+				{
+					continue;
+				}
+
+				AController* PlayerController = Cast<AController>(FlowController->GetOwner());
+				ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
+
+				checkf(PlayerCharacter, TEXT("Client did not receive a TestMovementCharacter"));
+
+				if (FlowController->ControllerInstanceId == 1)
+				{
+					PlayerCharacter->SetActorLocation(FVector(0.0f, 0.0f, 50.0f));
+				}
+				else
+				{
+					PlayerCharacter->SetActorLocation(FVector(100.0f, 300.0f, 50.0f));
```

Comment:
This will place all clients' characters above 1 in the same place, maybe causing issues with physics. Could just do a quick offset by instance id.

Reply:
Oh, yes that is correct, will change it now.